pkg/utils/secrets/static_token.go

```diff
@@ -114,6 +114,26 @@ func (s *StaticTokenSecretConfig) AppendStaticToken(staticToken *StaticToken) (*
 	return staticToken, nil
 }
 
+// RemoveFromStaticToken removes outdated tokens from the existing static tokens list.
+func RemoveFromStaticToken(staticToken *StaticToken, outdatedTokenKeys []string) *StaticToken {
```

Comment:
```suggestion
func (s *StaticToken) RemoveTokens(outdatedTokenUsernames ...string) *StaticToken {
```

Reply:
Please try to not make the assumption that a token key is always the username. This might not necessarily be the case.

Reply:
I saw that we have userID and userName always set alike so used the name, anyways I changed it.

Reply:
Thanks. FYI: I've ran into problems as part of https://github.com/gardener/gardener/pull/2478 and needed to refactor this section because this assumption broke in my case.

Reply:
so would the userID be enough from you PoV?

Reply:
I think the username or userid should be the "id" of such an entry, yes.